File: server/Models/Matchs/Requests/JoinLobbyRequest.cs
using System.Text.Json.Serialization;
using Models.Matchs.Enums;

namespace Models.Matchs.Requests
{
    public class JoinLobbyRequest
    {
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public BattleType BattleType { get; set; }
        public int NumberOfPlayers { get; set; }

        [JsonConverter(typeof(JsonStringEnumConverter))]
        public MatchContentType ContentType { get; set; }
        public int? TopicId { get; set; }
    }
}

File: mobile/app/src/main/java/com/n7/quizbattle/lobby/models/JoinLobbyRequest.java
package com.n7.quizbattle.lobby.models;

public class JoinLobbyRequest {
    private String battleType;
    private int numberOfPlayers;
    private String contentType;
    private Integer topicId;

    public JoinLobbyRequest(Integer topicId, String contentType, int numberOfPlayers, String battleType) {
        this.topicId = topicId;
        this.contentType = contentType;
        this.numberOfPlayers = numberOfPlayers;
        this.battleType = battleType;
    }
}
